pkg/controller/mutationtemplate/mutationtemplate_controller.go

```diff
@@ -124,30 +127,104 @@ func (r *ReconcileMutationTemplate) Reconcile(request reconcile.Request) (reconc
 		if errors.IsNotFound(err) {
 			// Object not found, return.  Created objects are automatically garbage collected.
 			// For additional cleanup logic use finalizers.
+			log.Info("didn't fine the instance - finalizer logic here?")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		log.Info("some other error occurred in getting the instance, requeuing")
+		return reconcile.Result{Requeue: true}, err
 	}
 	log.Info("Reconciling", "MutationTemplate", instance)
 
 	crd, err := r.opa.CreateMutationCRD(context.Background(), instance)
 	if err != nil {
-		log.Info("CreateMutationCRD failed, apparently.", "error", fmt.Sprintf("%v",err))
+		log.Info("CreateMutationCRD failed, apparently.", "error", fmt.Sprintf("%v", err))
+		return reconcile.Result{}, nil
+	}
+
+	name := crd.GetName()
+	namespace := crd.GetNamespace()
+	if instance.GetDeletionTimestamp().IsZero() {
```

Comment:
Maybe a comment here: "The MutationTemplate isn't being deleted, so create the underlying CRD if it doesn't exist"